Source/WebKit/Platform/IPC/StreamServerConnection.cpp

```diff
@@ -68,22 +51,23 @@ StreamServerConnection::~StreamServerConnection()
     ASSERT(!m_connection->isValid());
 }
 
-void StreamServerConnection::open()
+void StreamServerConnection::open(StreamConnectionWorkQueue& workQueue)
 {
-    static LazyNeverDestroyed<DedicatedConnectionClient> s_dedicatedConnectionClient;
-    static std::once_flag s_onceFlag;
-    std::call_once(s_onceFlag, [] {
-        s_dedicatedConnectionClient.construct();
-    });
+    m_workQueue = &workQueue;
     // FIXME(http://webkit.org/b/238986): Workaround for not being able to deliver messages from the dedicated connection to the work queue the client uses.
     m_connection->addMessageReceiveQueue(*this, { });
-    m_connection->open(s_dedicatedConnectionClient.get());
+    m_connection->open(*this, *m_workQueue);
 }
 
 void StreamServerConnection::invalidate()
 {
+    if (!m_workQueue)
+        return;
     m_connection->removeMessageReceiveQueue({ });
     m_connection->invalidate();
+    m_workQueue = nullptr;
+    Locker locker { m_outOfStreamMessagesLock };
+        if (m_outOfStreamMessages.isEmpty())
```

Comment:
It looks like there is a statement missing after the conditional or should this be an assertion?